code/obj/item/device/pda2/base_os.dm

```diff
@@ -292,6 +285,75 @@
+							var/datum/ringtone/rt = null
+							var/rtButton = "Default"
+							var/muteButton = "<a href='byond://?src=\ref[src];manageBlock=["add"];type=["mailgroup"];entry=[mailgrp]'>Block</a>"
+							var/leaveButton = "<a href='byond://?src=\ref[src];message_func=leave_group;groupname=[mailgrp]'>Leave</a>"
+							var/sendButton = ""
+							if(!src.clipboard?.dont_copy && src.master.fileshare_program)
+								sendButton = "<a href='byond://?src=\ref[src];input=send_file;group=[mailgrp]'>File</a>"
+							var/msgButton = "<a href='byond://?src=\ref[src];input=message;target=[mailgrp];department=1'>Mail</a>"
+							if((mailgrp in src.master.mailgroup_ringtones) && istype(src.master.mailgroup_ringtones[mailgrp], /datum/ringtone))
+								rt = src.master.mailgroup_ringtones[mailgrp]
+								rtButton = "[rt.name]"
+							if(mailgrp in src.muted_mailgroups)
+								muteButton = "<a href='byond://?src=\ref[src];manageBlock=["remove"];type=["mailgroup"];entry=[mailgrp]'>Unblock</a>"
+							if(mailgrp in src.master.reserved_mailgroups)
+								leaveButton = ""
+							. += "<tr><td>[mailgrp]</td><td>[rtButton]</td><td>[msgButton]</td><td>[sendButton]</td><td>[muteButton]</td><td>[leaveButton]</td></tr>"
+						. += "</table>"
+						. += "<hr><br>"
+						. += "<h4>Alert Settings</h4>"
+						. += "<table cellspacing=5>"
+
+						for(var/alert in src.master.alertgroups)
+							var/datum/ringtone/rt = null
+							var/rtButton = "Default"
+							var/muteButton = "<a href='byond://?src=\ref[src];manageBlock=["add"];type=["mailgroup"];entry=[alert]'>Mute</a>"
+							if(istype(src.master.alert_ringtones[alert], /datum/ringtone))
+								rt = src.master.alert_ringtones[alert]
+								rtButton = "[rt.name]"
+							if(alert in src.muted_mailgroups)
+								muteButton = "<a href='byond://?src=\ref[src];manageBlock=["remove"];type=["mailgroup"];entry=[alert]'>Unmute</a>"
+							. += "<tr><td>[alert]</td><td>[rtButton]</td><td>[muteButton]</td></tr>"
+
+				if(MODE_ADDRESSBOOK) // Specific names sent to us, also ringtones
+					. += "<h4>SpaceMessenger V4.0.5</h4>"
+					. += "<a href='byond://?src=\ref[src];mode=[MODE_MESSAGE]'>Back</a><br>"
+					. += "<h4>Address Book</h4><br>"
+					if(length(src.all_callers) < 1)
+						. += "Address book is empty!"
+					else
+						. += "<table cellspacing=5>"
+						for(var/caller in src.all_callers)
+							var/muteButton = "<a href='byond://?src=\ref[src];manageBlock=["add"];type=["single"];entry=[src.all_callers[caller]]'>Block</a>"
+							var/callButton = "<a href='byond://?src=\ref[src];input=message;target=[caller]'>Msg</a>"
+							var/sendButton = "<a href='byond://?src=\ref[src];input=send_file;target=[caller]'>Send</a>"
+							var/delButton = "<a href='byond://?src=\ref[src];delAddress=[caller]'>Del</a>"
+							if(caller in src.blocked_numbers)
+								muteButton = "<a href='byond://?src=\ref[src];manageBlock=["remove"];type=["single"];entry=[src.all_callers[caller]]'>Unblock</a>"
```

Comment:
Not sure why but this doesn't work.

Reply:
It was comparing an address to a list of names. Now, it compares the right values and should work just fine!